polygeist/tools/cgeist/Lib/CGExpr.cc

```diff
@@ -1765,144 +1770,14 @@ ValueCategory MLIRScanner::VisitCastExpr(CastExpr *E) {
     });
     return ValueCategory(Lres, /*isReference*/ false);
   }
-  case clang::CastKind::CK_IntegralToFloating: {
-    auto Scalar = Visit(E->getSubExpr()).getValue(Builder);
-    auto Ty = Glob.getTypes().getMLIRType(E->getType()).cast<mlir::FloatType>();
-    bool SignedType = true;
-    if (const auto *Bit =
-            dyn_cast<clang::BuiltinType>(&*E->getSubExpr()->getType())) {
-      if (Bit->isUnsignedInteger())
-        SignedType = false;
-      if (Bit->isSignedInteger())
-        SignedType = true;
-    }
-    if (SignedType)
-      return ValueCategory(
-          Builder.create<mlir::arith::SIToFPOp>(Loc, Ty, Scalar),
-          /*isReference*/ false);
-
-    return ValueCategory(Builder.create<mlir::arith::UIToFPOp>(Loc, Ty, Scalar),
-                         /*isReference*/ false);
-  }
-  case clang::CastKind::CK_FloatingToIntegral: {
-    auto Scalar = Visit(E->getSubExpr()).getValue(Builder);
-    auto Ty =
-        Glob.getTypes().getMLIRType(E->getType()).cast<mlir::IntegerType>();
-    bool SignedType = true;
-    if (const auto *Bit = dyn_cast<clang::BuiltinType>(&*E->getType())) {
-      if (Bit->isUnsignedInteger())
-        SignedType = false;
-      if (Bit->isSignedInteger())
-        SignedType = true;
-    }
-    if (SignedType)
-      return ValueCategory(
-          Builder.create<mlir::arith::FPToSIOp>(Loc, Ty, Scalar),
-          /*isReference*/ false);
-    return ValueCategory(Builder.create<mlir::arith::FPToUIOp>(Loc, Ty, Scalar),
-                         /*isReference*/ false);
-  }
-  case clang::CastKind::CK_IntegralCast: {
-    auto Scalar = Visit(E->getSubExpr()).getValue(Builder);
-    assert(Scalar);
-    auto PostTy =
-        Glob.getTypes().getMLIRType(E->getType()).cast<mlir::IntegerType>();
-    if (Scalar.getType().isa<mlir::LLVM::LLVMPointerType>())
-      return ValueCategory(
-          Builder.create<mlir::LLVM::PtrToIntOp>(Loc, PostTy, Scalar),
-          /*isReference*/ false);
-    if (Scalar.getType().isa<mlir::IndexType>() ||
-        PostTy.isa<mlir::IndexType>())
-      return ValueCategory(
-          Builder.create<arith::IndexCastOp>(Loc, PostTy, Scalar), false);
```

Comment:
This should be unreachable: integral values should not be represented as index values.